File: src/app/modules/m1-school-office/data-access/models/school-operational-budget-log.ts
import type { RecordStatus } from './common';

export interface SchoolOperationalBudgetLog {
  id: number;
  directorateId: number | null;
  schoolId: number | null;
  fiscalYear: string;
  budgetCategoryCode: string;
  categoryNameAr: string;
  allocatedAmount: number;
  consumedAmount: number;
  remainingAmount: number;
  status: RecordStatus;
  categoryNameEn: string | null;
  quarterNumber: number;
  approvedByDirectorId: number | null;
  lastTransactionDate: string | null;
  notesDescription: string | null;
}

export type CreateSchoolOperationalBudgetLogDto = Omit<SchoolOperationalBudgetLog, 'id' | 'status' | 'approvedByDirectorId'>;

export type UpdateSchoolOperationalBudgetLogDto = Omit<SchoolOperationalBudgetLog, 'directorateId' | 'schoolId' | 'status' | 'approvedByDirectorId'>;

File: 01_Backend/EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/SchoolOperationalBudgetLog.cs
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

public class SchoolOperationalBudgetLog : BaseAuditableEntity
{
    public long? DirectorateId { get; set; }
    public long? SchoolId { get; set; }
    public string FiscalYear { get; set; } = string.Empty;
    public string BudgetCategoryCode { get; set; } = string.Empty;
    public string CategoryNameAr { get; set; } = string.Empty;
    public decimal AllocatedAmount { get; set; }
    public decimal ConsumedAmount { get; set; }
    public decimal RemainingAmount { get; set; }
    public int Status { get; set; } // 1=Allocated, 2=InUse, 3=Exhausted, 4=Closed
    public string? CategoryNameEn { get; set; }
    public int QuarterNumber { get; set; } = 1;
    public long? ApprovedByDirectorId { get; set; }
    public DateTime? LastTransactionDate { get; set; }
    public string? NotesDescription { get; set; }

    // Navigation Properties
    public virtual Directorate? Directorate { get; set; }
    public virtual School? School { get; set; }
}
